crates/ruff_linter/src/rules/ruff/rules/unnecessary_round.rs

```diff
@@ -0,0 +1,189 @@
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{AlwaysFixableViolation, Applicability, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Arguments, Expr, ExprCall};
+use ruff_python_semantic::analyze::type_inference::{NumberLike, PythonType, ResolvedPythonType};
+use ruff_python_semantic::analyze::typing;
+use ruff_text_size::Ranged;
+
+/// ## What it does
+/// Checks for `round()` calls that have no effect on the input.
+///
+/// ## Why is this bad?
+/// Rounding a value that's already an integer is unnecessary.
+/// It's more clear to use the value directly.
+///
+/// ## Example
+///
+/// ```python
+/// a = round(1, 0)
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// a = 1
+/// ```
+#[derive(ViolationMetadata)]
+pub(crate) struct UnnecessaryRound;
+
+impl AlwaysFixableViolation for UnnecessaryRound {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Value being rounded is already an integer".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        "Remove unnecessary `round` call".to_string()
+    }
+}
+
+/// RUF057
+pub(crate) fn unnecessary_round(checker: &mut Checker, call: &ExprCall) {
+    let arguments = &call.arguments;
+
+    if !checker.semantic().match_builtin_expr(&call.func, "round") {
+        return;
+    }
+
+    let Some((rounded, rounded_kind, ndigits_kind)) = rounded_and_ndigits(checker, arguments)
+    else {
+        return;
+    };
+
+    if matches!(ndigits_kind, NdigitsKind::Other) {
+        return;
+    }
+
+    let applicability = match (rounded_kind, ndigits_kind) {
+        (RoundedKind::Int(InferredType::Equivalent), _) => Applicability::Safe,
```

Comment:
Some examples will help future readers

```suggestion
				// ```python
        // round(3)
        // round(3, 0)
        // round(3, ..)
        // ```
        (RoundedKind::Int(InferredType::Equivalent), _) => Applicability::Safe,
```